src/components/HTMLEngineProvider/HTMLRenderers/PreRenderer/index.js

```diff
@@ -1,70 +1,71 @@
-        return Math.abs(event.deltaY) > Math.abs(event.deltaX) * 2;
-    }
+    const isScrollingVertically = useCallback(
+        (event) =>
+            // Mark as vertical scrolling only when absolute value of deltaY is more than the double of absolute
+            // value of deltaX, so user can use trackpad scroll on the code block horizontally at a wide angle.
+            Math.abs(event.deltaY) > Math.abs(event.deltaX) * 2,
+        [],
+    );
+
+    const debouncedIsScrollingVertically = useCallback((event) => _.debounce(isScrollingVertically(event), 100, true), [isScrollingVertically]);
 
     /**
      * Manually scrolls the code block if code block horizontal scrollable, then prevents the event from being passed up to the parent.
      * @param {Object} event native event
      */
-    scrollNode(event) {
-        const node = this.ref.getScrollableNode();
-        const horizontalOverflow = node.scrollWidth > node.offsetWidth;
-        const isScrollingVertically = this.debouncedIsScrollingVertically(event);
-        if (event.currentTarget === node && horizontalOverflow && !isScrollingVertically) {
-            node.scrollLeft += event.deltaX;
-            event.preventDefault();
-            event.stopPropagation();
+    const scrollNode = useCallback(
+        (event) => {
+            const node = scrollViewRef.current.getScrollableNode();
+            const horizontalOverflow = node.scrollWidth > node.offsetWidth;
+            if (event.currentTarget === node && horizontalOverflow && !debouncedIsScrollingVertically(event)) {
+                node.scrollLeft += event.deltaX;
+                event.preventDefault();
+                event.stopPropagation();
+            }
+        },
+        [debouncedIsScrollingVertically],
+    );
+
+    useEffect(() => {
+        if (!scrollViewRef.current) {
+            return;
         }
-    }
+        scrollViewRef.current.getScrollableNode().addEventListener('wheel', scrollNode);
+        const eventListenerRefValue = scrollViewRef.current;
+
+        return () => {
+            eventListenerRefValue.getScrollableNode().removeEventListener('wheel', scrollNode);
+        };
+    }, [scrollNode]);
 
-    render() {
-        return (
-            <BasePreRenderer
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...this.props}
-                ref={(el) => (this.ref = el)}
-                onPressIn={() => DeviceCapabilities.canUseTouchScreen() && ControlSelection.block()}
-                onPressOut={() => ControlSelection.unblock()}
-            />
-        );
-    }
+    return (
+        <BasePreRenderer
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
+            ref={scrollViewRef}
+            onPressIn={() => DeviceCapabilities.canUseTouchScreen() && ControlSelection.block()}
+            onPressOut={ControlSelection.unblock()}
```

Comment:
It should be a callback.
```suggestion
            onPressOut={ControlSelection.unblock}
```